File: src/easy/Merge.java
package easy;

public class Merge {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        while (m - 1 >= 0 && n - 1 >= 0) {
            if (nums1[m - 1] >= nums2[n - 1]) {
                nums1[m + n - 1] = nums1[m - 1];
                m--;
            } else {
                nums1[m + n - 1] = nums2[n - 1];
                n--;
            }
        }
        while (n - 1 >= 0) {
            nums1[n - 1] = nums2[n - 1];
            n--;
        }
    }
}

File: csharp/LeetCode/Easy/Merge.cs
namespace LeetCode.Easy
{
    public partial class Solution
    {
        public void Merge(int[] nums1, int m, int[] nums2, int n)
        {
            var i = m - 1;
            var j = n - 1;
            var k = nums1.Length - 1;
            while (i >= 0 && j >= 0)
            {
                if (nums1[i] >= nums2[j])
                {
                    nums1[k] = nums1[i];
                    i--;
                }
                else
                {
                    nums1[k] = nums2[j];
                    j--;
                }
                k--;
            }
            while (j >= 0)
            {
                nums1[j] = nums2[j];
                j--;
            }
        }
    }
}
